service/extensions/extensions.go

```diff
@@ -83,6 +84,17 @@ func (bes *Extensions) NotifyPipelineNotReady() error {
 	return errs
 }
 
+func (bes *Extensions) NotifyConfigResolved(ctx context.Context, conf *confmap.Conf) error {
```

Comment:
Instead of doing this design where extension defines an optional interface to be implemented, we define in the service an optional interface for the Host, and extensions that need this can check if the Host passed at Start time implements that interface and uses it? What do you think?

Reply:
I think the advantage of defining an optional interface for Hosts would be that the configuration would be available during extension creation time, as where an extension interface would only provide the configuration to the extension in a subsequent method call. My goal has been to leave a door open for components to change during the lifetime of the Host, which I think an extension interface provides for while not complicating the design too much. If we don't want to take that into consideration at this point or we'd like to delay that decision and make breaking changes later, then I think an interface for the Host will work.